fusion-engine-driver/core/fusion_engine_node.cc

```diff
@@ -35,6 +35,7 @@ FusionEngineNode::FusionEngineNode()
     if (argValue == "tcp") {
       fe_interface_.initialize(this, this->get_parameter("tcp_ip").as_string(),
                                this->get_parameter("tcp_port").as_int());
+      dataListenerService();
```

Comment:
Are you sure this shouldn't be threaded like below:
```
      listener_thread_ =
          std::thread(std::bind(&FusionEngineNode::dataListenerService, this));

```

Reply:
In the other case (tty), the thread is needed so that RTCM messages are not lost and NMEA messages can be received regularly on a first thread. The role of the second thread is to retrieve the information sent serially by the card. In the case of tcp, it is only the retrieval of messages from the server and their recovery to send them to ros that is important. So I don't think a second thread is necessary.

Reply:
:+1:

Reply:
Thank you for the fix :+1: